src/dmd/dsymbolsem.d

```diff
@@ -636,6 +636,69 @@ package bool allowsContractWithoutBody(FuncDeclaration funcdecl)
     return true;
 }
 
+Expression resolveAliasThis(Scope* sc, Expression e, bool gag = false)
```

Comment:
I think that it is a better fit to move resolveAliasThis to `expressionsem.d`. There are no calls to this function in `dsymbolsem.d` and the operations are performed on an expressions, so technically it could be seen as been part of semantic on an expression.

Also, putting it in `expressionsem` will also make some of the imports to dsymbolsem that you added unnecessary.